### PR TITLE
Add instructions for CotEditor to `associating-text-editors-with-git.md`

### PR DESCRIPTION
**Pull request by noah-nuebling**
### Why:

CotEditor is an awesome text editor for macOS. It's super lightweight and powerful and really good to use with git.
I couldn't find any instructions on how to use it with git on the internet so I'm creating this, because it would have been very handy for me.

### What's being changed:

I added a section in the macOS tutorials on how to use git with the awesome CotEditor.

### Check off the following:
- [x] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Comment by noah-nuebling** [comment]
While reviewing the style guide, I've noticed two potential issues with the existing content of `associating-text-editors-with-git.md`.

1. The wording "shell utility" is used several times. `content-style-guide.md` recommends you use "terminal" and avoid "shell". I used the wording "`cot` command-line tool" in my addition, because that's the official product name in this case. Using "command-line tool" elsewhere in the document might be appropriate. To me, it seems to be a more common wording than "shell utility".
2. I used the "For more information, see X [or "Page/article title"] in the X documentation." wording recommended for links by `content-style-guide.md` in my addition, in order to stay consistent with the existing content of `associating-text-editors-with-git.md`. However I feel like this phrasing is a bit wordy and redundant in the case of my addition.
    - To demonstrate, I think that having the second instruction of my addition be:
        2. Install the "[`cot` command-line tool](https://coteditor.com/cot)".
    instead of:
        2. Install the `cot` command-line tool. For more information, see the "[`cot` command-line tool](https://coteditor.com/cot)" website.
     would be more straight forward and just as clear to users.
    Similar changes might be appropriate elsewhere in `associating-text-editors-with-git.md`.

**Comment by janiceilene** [comment]
@noah-nuebling Thanks so much for opening a PR! I'll get this triaged for review ⚡

**Comment by shati-patel** [comment]
👋  Hi @noah-nuebling, thanks for opening a PR! We aren't looking to add any more example editors right now, so I'm closing this PR. Thanks also for taking the time to review our style guide: if you have suggestions that would make our docs clearer, feel free to open a new issue with that information!

You can also take a look at the [contributing.md](https://github.com/github/docs/blob/main/CONTRIBUTING.md/) for more info about contributing to this repo. You can open a [discussion](https://github.com/github/docs/discussions) if you have any questions or check out issues with the [help wanted](https://github.com/github/docs/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) label. :sparkles:

Thank you again for your passion and interest in improving the documentation! 😊

**Comment by noah-nuebling** [comment]
Alright, thank you guys for taking the time to review my PR and I hope you're having a good week!
